x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelDefinition.java

```diff
@@ -129,21 +134,21 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         TrainedModelDefinition that = (TrainedModelDefinition) o;
         return Objects.equals(trainedModel, that.trainedModel) &&
-            Objects.equals(input, that.input) &&
-            Objects.equals(preProcessors, that.preProcessors);
+            Objects.equals(preProcessors, that.preProcessors) &&
+            Objects.equals(modelId, that.modelId);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(trainedModel, input, preProcessors);
+        return Objects.hash(trainedModel, preProcessors, modelId);
     }
 
     public static class Builder {
 
         private List<PreProcessor> preProcessors;
         private TrainedModel trainedModel;
+        private String modelId;
         private boolean processorsInOrder;
-        private Input input;
 
         private static Builder builderForParser() {
```

Comment:
I think this is not used anywhere.

Reply:
@dimitris-athanasiou it should be :). I will fix that promptly.